frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,484 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { safeLoad } from 'js-yaml';
+import { Helmet } from 'react-helmet';
+import { match } from 'react-router-dom';
+import { ClusterServiceVersionLogo, iconFor } from './index';
+import {
+  Alert,
+  Bullseye,
+  Button,
+  Card,
+  CardBody,
+  Flex,
+  FlexItem,
+  FlexModifiers,
+  Spinner,
+} from '@patternfly/react-core';
+import {
+  Firehose,
+  FirehoseResourcesResult,
+  ResourceLink,
+} from '@console/internal/components/utils';
+import {
+  groupVersionFor,
+  k8sUpdate,
+  referenceForModel,
+  referenceForGroupVersionKind,
+} from '@console/internal/module/k8s';
+import {
+  ClusterServiceVersionModel,
+  InstallPlanModel,
+  PackageManifestModel,
+  SubscriptionModel,
+} from '../models';
+import {
+  CRDDescription,
+  ClusterServiceVersionKind,
+  SubscriptionKind,
+  InstallPlanKind,
+  PackageManifestKind,
+} from '../types';
+import {
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared/src/components/status/icons';
+import { InstallPlanPreview } from './install-plan';
+import { RedExclamationCircleBadge } from '@console/shared/src/components/badges/RedExclamationCircleBadge';
+
+const getCRDFromObject = (obj: ClusterServiceVersionKind | InstallPlanKind) => {
+  return obj?.metadata?.annotations?.['operatorframework.io/initialization-resource'];
+};
+
+const isCRDAvailable = (obj: ClusterServiceVersionKind | InstallPlanKind) =>
+  typeof getCRDFromObject(obj) !== 'undefined';
+
+const installPageResources = (namespace: string) => [
+  {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    namespaced: true,
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('csvName')}`,
+    isList: true,
+    namespace,
+    prop: 'clusterServiceVersions',
+  },
+  {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('csvName')}`,
+    isList: true,
+    namespace: 'openshift-operators',
+    prop: 'allClusterServiceVersions',
+  },
+  {
+    kind: referenceForModel(SubscriptionModel),
+    namespaced: false,
```

Comment:
Is the intent to watch all Subscriptions in the cluster? What if different subscriptions in different namespaces have the same name?